Automattic/wp-calypso
# Analytics: Add MS Clarity to WPcom

Closes Automattic/martech#2229

## Proposed Changes

* Loads the base tracking code for MS Clarity on WPcom. This should also automatically trigger a page view. If you're testing via the live link below, you can also set this via URL parameters like `<URL>?flags=ad-tracking,cookie-banner`.

Essentially, this code is equivalent to the one MS provides us:
```
<script type="text/javascript">
    (function(c,l,a,r,i,t,y){
        c[a]=c[a]||function(){(c[a].q=c[a].q||[]).push(arguments)};
        t=l.createElement(r);t.async=1;t.src="https://www.clarity.ms/tag/"+i;
        y=l.getElementsByTagName(r)[0];y.parentNode.insertBefore(t,y);
    })(window, document, "clarity", "script", "j0cc1i1dba");
</script>
```

But it adds an init guard to prevent an undefined `window.clarity`, "calypso-ifies" it a bit, and categorizes this as an `advertising` tracker under GDPR/granular consent.

## Testing Instructions


* Set the flags `ad-tracking` and `cookie-banner` to true in `development.json`. Spin up Calypso locally, accept the GDPR banner if prompted.
* Go to any route in Calypso, e.g. the signup flow (`/start`). Observe in the network tab of the dev tools:
    * We load the initial Clarity script via `https://www.clarity.ms/tag/j0cc1i1dba`
    * This loads `https://www.clarity.ms/s/0.7.10/clarity.js`
    * Following this, you'll see multiple POST requests to `https://e.clarity.ms/collect`
* You should see no console errors related to this code.
* `window.clarity` is defined.
![Screenshot 2023-09-25 at 14 01 58](https://github.com/Automattic/wp-calypso/assets/52675688/567d4e75-af3a-4654-80d5-5484c926e244)
* If you have access to MS Clarity, you should be able to see your page view pretty quickly after.

* Delete the cookie called `sensitive_pixel_options` (related to tracking preferences) if you're in GDPR. 
    * Clarity will not load, and no console errors should be seen.
* Accept the cookie banner again, but do not include `advertising` as a bucket. If you're not in GDPR, you can set the `sensitive_pixel_options` cookie manually to `{"ok":true,"buckets":{"essential":true,"analytics":true,"advertising":false}}`
    * Clarity will not load—no console errors.   

## Pre-merge Checklist
